fusor-ember-cli/app/models/deployment-plan.js

```diff
@@ -3,133 +3,22 @@ import DS from 'ember-data';
-    }
-  }),
-
-  computeRoleCount: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Compute-1::count');
-    },
-    set(key, value) {
-      this.updateParam('Compute-1::count', value);
-    }
-  }),
-
-  controllerRoleFlavor: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Controller-1::Flavor');
-    },
-    set(key, value) {
-      this.updateParam('Controller-1::Flavor', value);
-    }
-  }),
-
-  controllerRoleCount: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Controller-1::count');
-    },
-    set(key, value) {
-      this.updateParam('Controller-1::count', value);
-    }
-  }),
-
-  cephStorageRoleFlavor: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Ceph-Storage-1::Flavor');
-    },
-    set(key, value) {
-      this.updateParam('Ceph-Storage-1::Flavor', value);
-    }
-  }),
-
-  cephStorageRoleCount: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Ceph-Storage-1::count');
-    },
-    set(key, value) {
-      this.updateParam('Ceph-Storage-1::count', value);
-    }
-  }),
-
-  cinderStorageRoleFlavor: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Cinder-Storage-1::Flavor');
-    },
-    set(key, value) {
-      this.updateParam('Cinder-Storage-1::Flavor', value);
-    }
-  }),
-
-  cinderStorageRoleCount: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Cinder-Storage-1::count');
-    },
-    set(key, value) {
-      this.updateParam('Cinder-Storage-1::count', value);
-    }
-  }),
-
-  swiftStorageRoleFlavor: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Swift-Storage-1::Flavor');
-    },
-    set(key, value) {
-      this.updateParam('Swift-Storage-1::Flavor', value);
-    }
-  }),
-
-  swiftStorageRoleCount: Ember.computed('parameters.@each.value', {
-    get(key) {
-      return this.getParamValue('Swift-Storage-1::count');
-    },
-    set(key, value) {
-      this.updateParam('Swift-Storage-1::count', value);
-    }
-  }),
-
-  getParam(paramName, paramsOverride) {
-    var params = paramsOverride || this.get('parameters');
-    return params ? params.findBy('id', paramName) : null;
+  getParam(fieldName) {
+    return this.get('parameters')[fieldName];
   },
 
-  getParamValue(paramName, paramsOverride) {
-    var param = this.getParam(paramName, paramsOverride);
-    return param ? param.get('value') : null;
+  getParamValue(fieldName) {
+    let param =  this.getParam(fieldName);
+    return param ? param.Default : undefined;
   },
 
-  updateParam(name, value) {
-    var param = this.getParam(name);
-
+  setParamValue(fieldName, value) {
+    let param =  this.getParam(fieldName);
     if (param) {
-      param.set('value', value);
+      param.Default = value;
```

Comment:
I think this code is actually unnecessary now since we don't update the deployment plan when updating role/count directly anymore and we use our db and sync them at the end.  I'll go ahead and remove the setParamValue and it's callers and test  it today.